cmd/account/iam-cred-rotate.go

```diff
@@ -0,0 +1,1578 @@
+	osdManagedAdminUsername := o.osdManagedAdminUsername
+	if osdManagedAdminUsername == "" {
+		osdManagedAdminUsername = common.OSDManagedAdminIAM + "-" + o.accountIDSuffixLabel
+	}
+	o.log.Debug(o.ctx, "Check if AWS user '%s' exists\n", osdManagedAdminUsername)
+	checkUser, err := o.awsClient.GetUser(&iam.GetUserInput{UserName: awsSdk.String(osdManagedAdminUsername)})
+	var nse *iamTypes.NoSuchEntityException
+	if (err != nil && errors.As(err, &nse)) || checkUser == nil {
+		o.log.Info(o.ctx, "User Not Found: '%s', trying user:'%s' instead...\n", osdManagedAdminUsername, common.OSDManagedAdminIAM)
+		osdManagedAdminUsername = common.OSDManagedAdminIAM
+		checkUser, err := o.awsClient.GetUser(&iam.GetUserInput{UserName: awsSdk.String(osdManagedAdminUsername)})
+		if err != nil {
+			return "", err
+		}
+		if checkUser == nil {
+			return "", fmt.Errorf("user Not Found:'%s'", o.osdManagedAdminUsername)
+		}
+	}
+	if err != nil {
+		return "", err
+	}
+	return osdManagedAdminUsername, nil
+}
+
+// Cli helper to print aws access key info and return
+func (o *rotateCredOptions) cliPrintOsdManagedAdminCreds() error {
+	osdManagedAdminUsername, err := o.checkOsdManagedUsername()
+	if err != nil {
+		return err
+	}
+	currKeys, clientInput, err := o.getKeyInfo(osdManagedAdminUsername, common.AWSAccountNamespace, o.secretName)
+	if err != nil {
+		return err
+	}
+	err = o.printKeyInfo(currKeys, clientInput, osdManagedAdminUsername)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Intended to save the new secret data in yaml format so a user can review and/or apply it from the CLI
+// This should only be called upon error to write/update the new secret with the new aws creds.
+func (o *rotateCredOptions) saveSecretYaml(secretNamespace string, secretName string, accessKey *iamTypes.AccessKey) (string, error) {
+	o.log.Warn(o.ctx, "Attempting to save secret yaml to file, ns:'%s', name:'%s'\n", secretNamespace, secretName)
+	if accessKey == nil {
+		err := fmt.Errorf("null AccessKey obj provided to saveSecretYaml")
+		o.log.Warn(o.ctx, "%v", err)
+		return "", err
+	}
+	secret := &corev1.Secret{}
+	err := o.hiveKubeClient.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: secretNamespace}, secret)
+	if err != nil {
+		o.log.Warn(o.ctx, "Failed to fetch existing secret: NS:'%s', NAME:'%s'. Err:'%s'", secretNamespace, secretName, err)
+		return "", err
+	}
+	newData := map[string][]byte{
+		"aws_user_name":         []byte(*accessKey.UserName),
+		"aws_access_key_id":     []byte(*accessKey.AccessKeyId),
+		"aws_secret_access_key": []byte(*accessKey.SecretAccessKey),
+	}
+	secret.Data = newData
+	s := k8json.NewYAMLSerializer(k8json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+	if s == nil {
+		err := fmt.Errorf("err saving secret to yaml. Failed to create k8 serializer")
+		o.log.Warn(o.ctx, "%v", err)
+		return "", err
+	}
+	saveFile, err := os.CreateTemp(os.TempDir(), string(*accessKey.UserName))
```

Comment:
Is it overkill to add a chmod 600?  Probably right?  Something about saving the secret locally makes me uncomfortable, but I dunno if that's reasonable or not. 

Lemme tractor-beam @wshearn for Team Security's take.

Reply:
Maybe in the event of an error like this it could be displayed on-screen instead?  I dunno, I'm probably being over-paranoid.

Reply:
Thanks for the input! I agree. Knowing the flow better, I dont think there's a lot of benefit to this upon failure either. I'll remove all of this. In terms of recovering from errors during the key rotation portion,  I've added some retry logic around the syncset portion (per potential syncset collisions noted in the original SOP).

Reply:
Removed in recent commit(s)